src/Voron/Platform/Posix/PosixHelper.cs

```diff
@@ -43,6 +43,7 @@ public static void AllocateFileSpace(StorageEnvironmentOptions options, int fd,
 
         public static unsafe void WriteFileHeader(FileHeader* header, VoronPathSetting path)
         {
+            bool fileExists = Syscall.access(path.FullPath, (int)(AccessMode.R_OK | AccessMode.W_OK)) == 0;
```

Comment:
Better to check here that the size is the expected size and skip fsync on dir otherwise